File: Lab6_OP_Bogun.cpp
#include <windows.h>
#include <iostream>
using namespace std;
double formula(double n, double k, double f)
{
	double p = ((n + k + f) / 2);
	double Sq = sqrt(p * (p - n) * (p - k) * (p - f));
	return Sq;
}

int main()
{
	SetConsoleCP(1251);
	SetConsoleOutputCP(1251);
	double a, b, c, a1, b1, c1, a2, b2, c2;

	do {
		cout << "Введіть довжини сторін трикутника\n а = "; cin >> a;
		cout << " b = "; cin >> b;
		cout << " с = "; cin >> c;
		
		if (((a + b) <= c) || ((a + c) <= b) || ((b + c) <= a)) 
			cout << "Сторони першого трикутника невірні. Трикутника із такими сторонами не існує. Спробуйте ще раз\n";
		else
		    cout << "Square = " << formula(a, b, c) << endl;
	} while ((a + b) <= c || (a + c) <= b || (b + c) <= a);


	do {
		cout << "Введіть довжини сторін другого трикутника\n а = "; cin >> a1;
		cout << " b = "; cin >> b1;
		cout << " с = "; cin >> c1;

		if (((a1 + b1) <= c1) || ((a1 + c1) <= b1) || ((b1 + c1) <= a1))
			cout << "Сторони другого трикутника невірні.Трикутника із такими сторонами не існує! Спробуйте ще раз\n";
		else
			cout << "Square = " << formula(a1, b1, c1) << endl;
	} while ((a1 + b1) <= c1 || (a1 + c1) <= b1 || (b1 + c1) <= a1);


	do {
		cout << "Введіть довжини сторін третього трикутника\n а = "; cin >> a2;
		cout << " b = "; cin >> b2;
		cout << " с = "; cin >> c2;

		if (((a2 + b2) <= c2) || ((a2 + c2) <= b2) || ((b2 + c2) <= a2))
			cout << "Сторони третього трикутника невірні.Трикутника із такими сторонами не існує! Спробуйте ще раз\n";
		else
			cout << "Square = " << formula(a2, b2, c2) << endl;
	} while ((a2 + b2) <= c2 || (a2 + c2) <= b2 || (b2 + c2) <= a2);


	if ((formula(a, b, c)) > (formula(a1, b1, c1)) && (formula(a, b, c)) > (formula(a2, b2, c2)))
		cout << "Перший трикутник найбільший";
	else if ((formula(a1, b1, c1)) > (formula(a, b, c)) && (formula(a1, b1, c1)) > (formula(a2, b2, c2)))
		cout << "Другий трикутник найбільший";
	else if ((formula(a2, b2, c2)) > (formula(a, b, c)) && (formula(a2, b2, c2)) > (formula(a1, b1, c1)))
		cout << "Третій трикутник найбільший";

	else
		cout << "Трикутники рівні" << endl;


}

File: Lab6_OP_Bogun.py
import math


def formula(n, k, f):
    p = float((n + k + f)/2)
    Sq = math.sqrt(p*(p-n)*(p-k)*(p-f))
    return Sq

def maxSq(Sq1, Sq2, Sq3):
    if ((Sq1 > Sq2) and (Sq1 > Sq3)):
        Max = 1
    if ((Sq2 > Sq1) and (Sq2 > Sq3)):
        Max = 2
    if ((Sq3 > Sq1) and (Sq3 > Sq2)):
        Max = 3
    return Max

def inputt():
    flag = True
    while flag:
        flag = False
        a = float(input(" а = "))
        b = float(input(" b = "))
        c = float(input(" c = "))
        if((a >= b + c) or (b >= a + c) or (c >= b + a)):
            print("Такого трикутника не існує")
            flag = True
    return a, b, c


print("1 трикутник:")
a, b, c = inputt()
print("2 трикутник:")
a1, b1, c1 = inputt()
print("3 трикутник:")
a2, b2, c2 = inputt()

Sq1 = formula(a, b, c)
Sq2 = formula(a1, b1, c1)
Sq3 = formula(a2, b2, c2)
print("Square 1 = ", Sq1)
print("Square 2 = ", Sq2)
print("Square 3 = ", Sq3)
print( maxSq(Sq1, Sq2, Sq3), " трикутник найбільший")


